cli/cmd/init.go

```diff
@@ -220,12 +220,47 @@ func validateStackNameAvailable(ctx context.Context, stackService *stackservice.
 			return nil
 		}
 
-		_, err := stackService.GetStack(ctx, stackName)
+		metas, err := stackService.CollectStackInfo(ctx, happyClient.HappyConfig.App(), true)
 		if err != nil {
+			return errors.Wrap(err, "unable to collect stack info")
+		}
+
+		for _, meta := range metas {
+			if meta.Stack == stackName {
+				if meta.AppName == happyClient.HappyConfig.App() {
+					return nil
```

Comment:
```suggestion
		for _, meta := range metas {
			if meta.Stack == stackName && meta.AppName == happyClient.HappyConfig.App(){
					return nil
```

fewer nested statements